pkgs/development/python-modules/torch/bin.nix

```diff
@@ -29,37 +34,65 @@ in buildPythonPackage {
 
   format = "wheel";
 
-  disabled = !(isPy38 || isPy39 || isPy310);
+  disabled = (pythonOlder "3.8") || (pythonAtLeast "3.12");
 
   src = fetchurl srcs."${stdenv.system}-${pyVerNoDot}" or unsupported;
 
   nativeBuildInputs = [
     addOpenGLRunpath
+    autoPatchelfHook
+    cudaPackages.autoAddOpenGLRunpathHook
     patchelf
   ];
 
+  buildInputs = with cudaPackages; [
+    # $out/${sitePackages}/nvfuser/_C*.so wants libnvToolsExt.so.1 but torch/lib only ships
+    # libnvToolsExt-$hash.so.1
+    cuda_nvtx
+  ];
+
+  autoPatchelfIgnoreMissingDeps = [
+    # This is the hardware-dependent userspace driver that comes from
+    # nvidia_x11 package. It must be deployed at runtime in
+    # /run/opengl-driver/lib or pointed at by LD_LIBRARY_PATH variable, rather
+    # than pinned in runpath
+    "libcuda.so.1"
+  ];
+
   propagatedBuildInputs = [
     future
     numpy
     pyyaml
     requests
     setuptools
     typing-extensions
+    sympy
+    jinja2
+    networkx
+    filelock
+  ] ++ lib.optionals stdenv.isx86_64 [
+    openai-triton
   ];
 
   postInstall = ''
     # ONNX conversion
     rm -rf $out/bin
   '';
 
-  postFixup = let
-    rpath = lib.makeLibraryPath [ stdenv.cc.cc.lib ];
-  in ''
-    find $out/${python.sitePackages}/torch/lib -type f \( -name '*.so' -or -name '*.so.*' \) | while read lib; do
-      echo "setting rpath for $lib..."
-      patchelf --set-rpath "${rpath}:$out/${python.sitePackages}/torch/lib" "$lib"
-      addOpenGLRunpath "$lib"
-    done
+  postFixup = ''
+    addAutoPatchelfSearchPath "$out/${python.sitePackages}/torch/lib"
+
+    patchelf $out/${python.sitePackages}/torch/lib/libcudnn.so.8 --add-needed libcudnn_cnn_infer.so.8
+
+    pushd $out/${python.sitePackages}/torch/lib || exit 1
+      for LIBNVRTC in ./libnvrtc*so
```

Comment:
Oh, the actual file seems to end with a version number: `libnvrtc-672ee683.so.11.2`. Maybe we make it just `./libnvrtc*`

Reply:
Done !